eth/stagedsync/stage_snapshots.go

```diff
@@ -233,7 +233,18 @@ func DownloadAndIndexSnapshotsIfNeed(s *StageState, ctx context.Context, tx kv.R
 		if err := snapshotsync.WaitForDownloader(ctx, s.LogPrefix(), cfg.historyV3, cstate, cfg.agg, tx, cfg.blockReader, &cfg.chainConfig, cfg.snapshotDownloader, s.state.StagesIdsList()); err != nil {
 			return err
 		}
+
+		if err := cfg.blockReader.Snapshots().ReopenFolder(); err != nil {
```

Comment:
WaitForDownloader does Reopen inside

Reply:
Ok - I'll check this - I hit an issue though - I need to re look at the tests and see what happens - it came after the close - see below for details as to why this is necessary.

Reply:
Fixed here: https://github.com/ledgerwatch/erigon/pull/9456/commits/b336e85c713e4005a1f54b9438956d4de2fca886